frontend/packages/dev-console/src/components/import/__tests__/deployImage-submit-utils.spec.ts

```diff
@@ -63,6 +64,215 @@ describe('DeployImage Submit Utils', () => {
     });
   });
 
+  describe('createOrUpdateImageStream', () => {
+    const k8sCreate = jest.spyOn(k8s, 'k8sCreate');
+    const k8sUpdate = jest.spyOn(k8s, 'k8sUpdate');
+    const k8sWaitForUpdate = jest.spyOn(k8s, 'k8sWaitForUpdate');
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+      k8sCreate.mockReturnValue(null);
+      k8sUpdate.mockReturnValue(null);
+      k8sWaitForUpdate.mockReturnValue(null);
+    });
+
+    afterAll(() => {
+      jest.restoreAllMocks();
+    });
+
+    it('should call only k8sCreate when call "dry-run create"', async () => {
+      // Mock data
+      const mockImageStreamData = {
+        apiVersion: 'image.openshift.io/v1',
+        kind: 'ImageStream',
+        metadata: {
+          labels: {
+            app: 'test-app',
+            'app.kubernetes.io/component': 'test-app',
+            'app.kubernetes.io/instance': 'test-app',
+            'app.kubernetes.io/part-of': 'mock-app',
+          },
+          name: 'test-app',
+          namespace: 'mock-project',
+        },
+        spec: {
+          tags: [
+            {
+              name: 'latest',
+              annotations: {
+                'openshift.io/generated-by': 'OpenShiftWebConsole',
+                'openshift.io/imported-from': 'myimage',
+              },
+              from: { kind: 'DockerImage', name: 'myimage' },
+              importPolicy: { insecure: false },
+              referencePolicy: { type: 'Local' },
+            },
+          ],
+        },
+      };
```

Comment:
nit: this data is used even in below it blocks , we can have it in separate mock data file or before each and use it (update only particular spec field if needed) WDYT?

Reply:
+:100: Absolutely. Extracted this into the mock file which contains also the form data. PTAL